activerecord/test/cases/associations_test.rb

```diff
@@ -164,6 +164,15 @@ def test_has_many_association_with_composite_foreign_key_loads_records
     assert_includes(comments, sharded_comments(:great_comment_blog_post_one))
   end
 
+  def test_has_many_association_from_a_non_qc_model_loads_records
+    blog_post = sharded_blog_posts(:great_post_blog_one)
+    blog_post = Sharded::BlogPostWithRevision.find(blog_post.id)
+
+    comments = blog_post.comments.to_a
+    assert_includes(comments, sharded_comments(:wow_comment_blog_post_one))
+    assert_includes(comments, sharded_comments(:great_comment_blog_post_one))
+  end
```

Comment:
Should we add assertions on the SQL used to select these to assert the expected columns are present in the query, and ones we don't expect aren't. If the way we fetch records changes, but the same records end up being loaded, it will go unnoticed.

Thoughts?